ddl/index.go

```diff
@@ -562,6 +565,9 @@ func (w *worker) onCreateIndex(d *ddlCtx, t *meta.Meta, job *model.Job, isPK boo
 			if kv.ErrKeyExists.Equal(err) || errCancelledDDLJob.Equal(err) || errCantDecodeRecord.Equal(err) {
 				logutil.BgLogger().Warn("[ddl] run add index job failed, convert job to rollback", zap.String("job", job.String()), zap.Error(err))
 				ver, err = convertAddIdxJob2RollbackJob(t, job, tblInfo, indexInfo, err)
+				if err1 := t.RemoveDDLReorgHandle(job, reorgInfo.elements); err1 != nil {
```

Comment:
it is handled differently with alter column, should move it before `ver, err = convertAddIdxJob2RollbackJob(t, job, tblInfo, indexInfo, err)`? And should it in alter column also be rollingback?

Reply:
I think needn't. `modify column` do `RemoveDDLReorgHandle`, then set `job.State = model.JobStateRollingback`. Therefore, it is ok to put `RemoveDDLReorgHandle` before or after `job.State = model.JobStateRollingback`.  But, in `add index` do `convertAddIdxJob2RollbackJob` maybe failed. So I put `RemoveDDLReorgHandle` after `convertAddIdxJob2RollbackJob`.